src/niquests/_typing.py

```diff
@@ -1,5 +1,6 @@
 from __future__ import annotations
 
+from pathlib import Path
```

Comment:
I am wondering if we should not use `os.PathLike` abstract definition instead.
And leverage the `__fspath__` method instead. So that we don't exclude other implementation that aren't pathlib. WDYT?